### PR TITLE
Enable Enzyme

### PR DESCRIPTION
**Pull request by Red-Portal**
* Update Enzyme external dependency to match the new AD interface
* Enable unit tests with Enzyme

Currently, this PR depends on upstream the following upstream issues:
* https://github.com/EnzymeAD/Enzyme.jl/issues/1548
* https://github.com/EnzymeAD/Enzyme/pull/1933
* https://github.com/EnzymeAD/Enzyme.jl/pull/1550

### CODE REVIEW

File: ext/AdvancedVIEnzymeExt.jl [hunk 1]
```diff
@@ -11,19 +11,46 @@ else
...
+    f,
+    x  ::AbstractVector{<:Real},
+    out::DiffResults.MutableDiffResult
+)
+    Enzyme.API.runtimeActivity!(true)
+    ∇x = DiffResults.gradient(out)
+    fill!(∇x, zero(eltype(∇x)))
+    _, y = Enzyme.autodiff(
+        Enzyme.ReverseWithPrimal,
+        f,
```
Comment (wsmoses) [inline ext/AdvancedVIEnzymeExt.jl:30]:
```suggestion
        Enzyme.Const(f),
```

**Comment by Red-Portal** [comment]
@wsmoses It seems Enzyme is hitting some C++-level errors on `1.7`. All tests do pass on `1.10` though. Is Enzyme not expected to work on `1.7` or should I try to pack a MWE just for `1.7`?

**Comment by wsmoses** [comment]
So 1.7 and below have known GC bugs in Julia so support is weaker than
1.10. That said it should work so an MWE is welcome (and feel free to merge
1.10 here in the interim and we can follow up with a 1.7 PR once a MWE is
opened and fixed).

File: test/runtests.jl [hunk 2]
```diff
@@ -47,7 +47,7 @@ end
 const PROGRESS = haskey(ENV, "PROGRESS")
 
 if GROUP == "All" || GROUP == "Inference"
-    include("inference/repgradelbo_distributionsad.jl")
+    #include("inference/repgradelbo_distributionsad.jl")
```
[the remark this answers is not in the document]
Comment (Red-Portal) [reply test/runtests.jl:50]: @yebai This was temporary to check if it was `DistributionAD` that was breaking on 1.7. But it was not the case. So I'll enable it back before merging. I think I'll run CI on both 1.10 and 1.7 and disable Enzyme when testing on 1.7 for now, as suggested by @wsmoses .

**Comment by Red-Portal** [comment]
@wsmoses Unfortunately, some of the tests on 1.10 are still failing, though I can't reproduce it locally. Seems like there is still some work left to do :disappointed:

**Comment by Red-Portal** [comment]
Hurray, all the tests finally passed! Thank you so much @wsmoses for the amazing real-time work!

**Pull request merged**